### PR TITLE
Add web/templates as a Qute template root

### PR DESCRIPTION
**Pull request by ia3andy**
cc @angelozerr

### CODE REVIEW

File: qute-components/deployment/src/main/java/io/quarkiverse/web/bundler/qute/components/deployment/WebBundlerQuteTemplateRootProcessor.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class WebBundlerQuteTemplateRootProcessor {
+
+    @BuildStep
+    void provideWebTemplateRoot(
+            BuildProducer<NativeImageResourceDirectoryBuildItem> nativeImageResourceDirectoryProducer,
+            BuildProducer<TemplateRootBuildItem> templateRootProducer,
+            WebBundlerConfig config) {
+        templateRootProducer.produce(new TemplateRootBuildItem(config.fromWebRoot("templates")));
+        nativeImageResourceDirectoryProducer
+                .produce(new NativeImageResourceDirectoryBuildItem(config.fromWebRoot("templates")));
```
[the remark this answers is not in the document]
Comment (ia3andy) [reply qute-components/deployment/src/main/java/io/quarkiverse/web/bundler/qute/components/deployment/WebBundlerQuteTemplateRootProcessor.java:18]: Maybe it should be done by the Qute Processor?

[the remark this answers is not in the document]
Comment (mkouba) [reply qute-components/deployment/src/main/java/io/quarkiverse/web/bundler/qute/components/deployment/WebBundlerQuteTemplateRootProcessor.java:18]: Well, we do produce a `NativeImageResourceBuildItem` for each template found. TBH I've never seen `NativeImageResourceDirectoryBuildItem` before.

Also we test a native image in `quarkus-qute-web`.

What's the actual problem?

[the remark this answers is not in the document]
Comment (ia3andy) [reply qute-components/deployment/src/main/java/io/quarkiverse/web/bundler/qute/components/deployment/WebBundlerQuteTemplateRootProcessor.java:18]: @mkouba the tests were not passing in native mode

[the remark this answers is not in the document]
Comment (mkouba) [reply qute-components/deployment/src/main/java/io/quarkiverse/web/bundler/qute/components/deployment/WebBundlerQuteTemplateRootProcessor.java:18]: `quarkus-web-bundler ` tests I assume? Is there anything special in those tests?

[the remark this answers is not in the document]
Comment (ia3andy) [reply qute-components/deployment/src/main/java/io/quarkiverse/web/bundler/qute/components/deployment/WebBundlerQuteTemplateRootProcessor.java:18]: @phillip-kruger I think this is the origin of #172, `NativeImageResourceDirectoryBuildItem` might want the directory to exist (which should not lead to a NPE anyway), so I need to add a check here and maybe a fix in Quarkus

**Pull request merged**